src/app_service/service/community/service.nim

```diff
@@ -194,11 +198,19 @@ QtObject:
             error "Received a membership request for an unknown community", communityId=membershipRequest.communityId
             continue
           var community = self.joinedCommunities[membershipRequest.communityId]
-          community.pendingRequestsToJoin.add(membershipRequest)
-          self.joinedCommunities[membershipRequest.communityId] = community
-          self.events.emit(SIGNAL_COMMUNITY_EDITED, CommunityArgs(community: community))
+          if (membershipRequest.state == RequestToJoinType.Pending.int):
+            community.pendingRequestsToJoin.add(membershipRequest)
+            self.joinedCommunities[membershipRequest.communityId] = community
+            self.events.emit(SIGNAL_COMMUNITY_EDITED, CommunityArgs(community: community))
+
+            self.events.emit(SIGNAL_NEW_REQUEST_TO_JOIN_COMMUNITY, CommunityRequestArgs(communityRequest: membershipRequest))
```

Comment:
Just to be sure, this line belongs into the if-case here?

Reply:
fixed :)